src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -249,95 +280,66 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
+        }
 
         if (fixed_columns) {
-            const r1c0Table = r1c0.querySelector('table') as HTMLElement;
-            const r1c1Table = r1c0.querySelector('table') as HTMLElement;
-
-            r1c0Table.style.width = getComputedStyle(r1c1Table).width;
-
-            const lastVisibleTd = r1c0.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
-
-            let it = 0;
-            let currentWidth = r1c0.getBoundingClientRect().width;
-            let lastWidth = currentWidth;
-
-            do {
-                lastWidth = currentWidth
+            const lastFixedTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
+            if (lastFixedTd) {
+                const lastFixedTdBounds = lastFixedTd.getBoundingClientRect();
+                const lastFixedTdRight = lastFixedTdBounds.right - r1c1.getBoundingClientRect().left;
 
                 // Force first column containers width to match visible portion of table
-                if (lastVisibleTd) {
-                    const r1c0FragmentBounds = r1c0.getBoundingClientRect();
-                    const lastTdBounds = lastVisibleTd.getBoundingClientRect();
-                    currentWidth = lastTdBounds.right - r1c0FragmentBounds.left;
-
-                    const width = `${currentWidth}px`;
-
-                    r0c0.style.width = width;
-                    r1c0.style.width = width;
+                r0c0.style.width = `${lastFixedTdRight}px`;
+                r1c0.style.width = `${lastFixedTdRight}px`;
+
+                if (!cycle) {
+                    // Force second column containers width to match visible portion of table
+                    const firstVisibleTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns + 1})`);
+                    if (firstVisibleTd) {
+                        const r1c1FragmentBounds = r1c1.getBoundingClientRect();
+                        const firstTdBounds = firstVisibleTd.getBoundingClientRect();
+
+                        const width = firstTdBounds.left - r1c1FragmentBounds.left;
+                        r0c1.style.marginLeft = `-${width}px`;
+                        r1c1.style.marginLeft = `-${width}px`;
+                    }
                 }
-
-                // Force second column containers width to match visible portion of table
-                const firstVisibleTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns + 1})`);
-                if (firstVisibleTd) {
-                    const r1c1FragmentBounds = r1c1.getBoundingClientRect();
-                    const firstTdBounds = firstVisibleTd.getBoundingClientRect();
-
-                    const width = firstTdBounds.left - r1c1FragmentBounds.left;
-                    r0c1.style.marginLeft = `-${width}px`;
-                    r0c1.style.marginRight = `${width}px`;
-                    r1c1.style.marginLeft = `-${width}px`;
-                    r1c1.style.marginRight = `${width}px`;
```

Comment:
`marginRight` caused the table to be smaller than expected with fixed columns. Removing it for both `r0c1` and `r1c1` fragments ensures the table is the expected size